Sources/Shared/Models/ConnectedApp.swift

```diff
@@ -0,0 +1,237 @@
+//
+//  ConnectedApp.swift
+//  VimeoNetworking
+//
+//  Copyright © 2019 Vimeo. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+/// An enumeration of the supported connected app types.
+/// - facebook: Represents a connection to Facebook.
+/// - linkedin: Represents a connection to LinkedIn.
+/// - twitter: Represents a connection to Twitter.
+/// - youtube: Represents a connection to YouTube.
+@objc public enum ConnectedAppType: Int {
+    case facebook
+    case linkedin
+    case twitter
+    case youtube
+
+    public var stringValue: String {
+        switch self {
+        case .facebook:
+            return String.facebook
+        case .linkedin:
+            return String.linkedin
+        case .twitter:
+            return String.twitter
+        case .youtube:
+            return String.youtube
+        }
+    }
+}
+
+///  An object that encapsulates the scopes necessary for interacting with features like publishing to social platforms
+///  or simulcasting a live stream.
+@objcMembers
+public class ConnectedAppScopes: VIMModelObject {
+
+    /// All scopes required for publishing to a specific social media platform.
+    public var publishToSocial: [String]?
+
+    /// All scopes required for simulcasting to a specific social media platform.
+    public var simulcast: [String]?
+
+    // MARK: - Overrides
+
+    public override func getObjectMapping() -> Any? {
+        return [
+            String.Key.publishToSocial: String.Value.publishToSocial
+        ]
+    }
+}
+
+/// A category that can be sent when publishing to a social media platform.
+@objcMembers
+public class PublishOptionItem: VIMModelObject {
+
+    /// The ID of the publish item.
+    public var identifier: String?
```

Comment:
Same re: public setters - I'm guessing the identifier should not be publicly writeable?